domain/infosync/info.go

```diff
@@ -122,7 +135,14 @@ func (is *InfoSyncer) GetServerInfoByID(ctx context.Context, id string) (*Server
 }
 
 // GetAllServerInfo gets all servers static information from etcd.
-func (is *InfoSyncer) GetAllServerInfo(ctx context.Context) (map[string]*ServerInfo, error) {
+func GetAllServerInfo(ctx context.Context) (map[string]*ServerInfo, error) {
+	if globalInfoSyncer == nil {
+		return nil, nil
```

Comment:
If `globalInfoSyncer` is nil, err shouldn't be nil. Otherwise it would panic when executing `serverInfo.IP`.

Reply:
Great.